Mathlib/Data/Finset/NatAntidiagonal.lean

```diff
@@ -35,6 +35,14 @@ instance instHasAntidiagonal : HasAntidiagonal ℕ where
   mem_antidiagonal {n} {xy} := by
     rw [mem_def, Multiset.Nat.mem_antidiagonal]
 
+lemma antidiagonal_eq_map (n : ℕ) :
+    antidiagonal n = (range (n + 1)).map ⟨fun i ↦ (i, n - i), fun _ _ h ↦ (Prod.ext_iff.1 h).1⟩ :=
+  rfl
+
+lemma antidiagonal_eq_image (n : ℕ) :
+    antidiagonal n = (range (n + 1)).image fun i ↦ (i, n - i) := by
+  simp only [antidiagonal_eq_map, map_eq_image, Function.Embedding.coeFn_mk]
```

Comment:
Is it worth adding the symmetric versions too, with `(n - i, i)`?

Also, is there a useful version of this that's true for some slightly more general locally finite orders?

Reply:
I can't generalise to locally finite orders because of the `n - i`. I did add the symmetric lemmas, though.